dig.go

```diff
@@ -168,10 +168,23 @@ type containerStore interface {
 	// Returns the providers that can produce values for the given group and
 	// type.
 	getGroupProviders(name string, t reflect.Type) []provider
+
+	// Finds and adds the node to the list of failed nodes of the graph in the
+	// container and mark the corresponding constructor and group as failed.
+	failGroupNodes(name string, t reflect.Type, id dot.CtorID)
```

Comment:
nit: consider updating the signature here to
```
failGroupNodes(t reflect.Type, group string, id dot.CtorID)
```

That way it is consistent with the `failNodes` signature (types first, then group / ids after).